### DISCUSSION TITLE
Lambda Context callbackWaitsForEmptyEventLoop

### DISCUSSION THREAD

**Discussion by masterbater** [honojs/hono#1519] [Q&A]
callbackWaitsForEmptyEventLoop  does this exist in lambda context, its untyped

Is this correct ? is there any better way to load or initiate database connection in lambda
```typescript
app.use("*", async (c, next) => {
  const { DATABASE_URL } = env<{ DATABASE_URL: string }>(c);

  const lambdaContext = c.env?.requestContext;
  if (lambdaContext) {
    lambdaContext.callbackWaitsForEmptyEventLoop = false;
  }
  await next();
});
```
Reference 
https://mongoosejs.com/docs/lambda.html

**Comment by yusukebe** [comment]
cc: @watany-dev

**Comment by watany-dev** [comment]
The `requestContext` refers to the common `event.requestContext` in Lambda. It might be resolved by providing `c.env.lambdaContext` in the following PR. 

https://github.com/honojs/hono/pull/1523

```ts
  app.get('/lambda-context', (c) => {
    const fnctx = c.env.lambdaContext
    console.log(fnctx..callbackWaitsForEmptyEventLoop)
    return c.json(fnctx)
  })
```

Although not directly related to this topic, the current access to requestContext can be confusing. I feel it would be preferable to bind the lambdaEvent and access it via c.env.lambdaEvent.requestContext.
